src/_pytest/capture.py

```diff
@@ -568,7 +568,7 @@ def __repr__(self):
             self.targetfd, getattr(self, "targetfd_save", None), self._state
         )
 
-    def start(self):
+    def start(self):  # type: ignore
```

Comment:
what's the type error here?

Reply:
I'd rather see `assert` guards for lateinit things -- that way we know in our tests when the assumption is violated and mypy can continue to typecheck the rest of the function -- I generally reserve `cast` / `# type: ignore` for actual mypy bugs only

Reply:
Not sure what you mean about `assert` guards, however I'll add a different workaround for this, to avoid the clash between a method name and the `__init__` initialization.

Reply:
for example:

```python
class C:
    def __init__(self) -> None:
        self.x = None  # type: Optional[int]

    def set_x(self, x: int) -> None:
        self.x = x

    def use_x(self) -> int:
        assert self.x is not None  # convince mypy that lateinit has occurred
        return self.x + 5
```

without the assertion:

```console
$ mypy t.py
t.py:13: error: Unsupported operand types for + ("None" and "int")
t.py:13: note: Left operand is of type "Optional[int]"
```

with the assert:

```console
$ mypy t.py
$
```

Reply:
Ah, right. The problem here is different ("foo is already defined").

Reply:
oh yikes! https://github.com/pytest-dev/pytest/blob/73767268eca69bb07bb71dfb4ab7b651295e0106/src/_pytest/capture.py#L546-L548